src/intents/executeDeclaration.ts

```diff
@@ -0,0 +1,36 @@
+import { TransactionData } from '@brinkninja/types'
+import SignedDeclaration from './SignedDeclaration'
+import evm from '../internal/EthereumJsVm'
+import { metaDelegateCall } from '../core'
+
+export type ExecuteIntentArgs = {
+  signedDeclaration: SignedDeclaration
+  intentIndex: number
+  unsignedCalls: string[]
+  deployAccount?: boolean
+}
+
+async function executeDeclaration ({
```

Comment:
I think we should name this `executeIntent()` because it's actually executing at the intent level. It's executing the intent at `intentIndex`, it just needs the signed declaration to validate that the intent was signed within that declaration